lib/Analysis/SecretnessAnalysis/SecretnessAnalysis.h

```diff
@@ -106,8 +106,53 @@ class SecretnessAnalysis
                                ArrayRef<SecretnessLattice *> results) override;
 };
 
+template <typename ChildAnalysis>
+class SecretnessAnalysisDependent {
+ private:
+  ChildAnalysis *getChildAnalysis() {
+    return static_cast<ChildAnalysis *>(this);
+  }
+
+ protected:
+  // this method is used inside an Analysis
+  bool ensureSecretness(Operation *op, Value value) {
```

Comment:
Can you add a few more words about what this actually guarantees to the comment? (e.g., how it deals with "unknown")

Reply:
Done